ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/subviews/ToolVolumesFromListEditor.java

```diff
@@ -0,0 +1,183 @@
+    private int unique_volumes_from_id;
+    private ListStoreEditor<ToolVolumesFrom> listStoreEditor;
+
+    @UiField
+    ListStore<ToolVolumesFrom> listStore;
+    @UiField(provided = true)
+    ToolAdminView.ToolAdminViewAppearance appearance;
+
+
+    @Inject
+    public ToolVolumesFromListEditor(ToolAdminView.ToolAdminViewAppearance appearance,
+                                     ToolVolumesFromProperties toolVolumesFromProperties) {
+        this.appearance = appearance;
+        listStore = new ListStore<>(new ModelKeyProvider<ToolVolumesFrom>() {
+            @Override
+            public String getKey(ToolVolumesFrom item) {
+                return getVolumesFromTag(item);
+            }
+        });
+        listStoreEditor = new ListStoreEditor<>(listStore);
+        listStore.setAutoCommit(true);
+
+        ColumnConfig<ToolVolumesFrom, String> name = new ColumnConfig<>(toolVolumesFromProperties.name(),
+                                                                        appearance.containerVolumesFromNameWidth(),
+                                                                        appearance.containerVolumesFromNameLabel());
+        ColumnConfig<ToolVolumesFrom, String> namePrefix = new ColumnConfig<>(toolVolumesFromProperties.namePrefix(),
+                                                                              appearance.containerVolumesFromNamePrefixWidth(),
+                                                                              appearance.containerVolumesFromNamePrefixLabel());
+        ColumnConfig<ToolVolumesFrom, String> tag = new ColumnConfig<>(toolVolumesFromProperties.tag(),
+                                                                       appearance.containerVolumesFromTagWidth(),
+                                                                       appearance.containerVolumesFromTagLabel());
+        ColumnConfig<ToolVolumesFrom, String> url = new ColumnConfig<>(toolVolumesFromProperties.url(),
+                                                                       appearance.containerVolumesFromURLWidth(),
+                                                                       appearance.containerVolumesFromURLLabel());
+        ColumnConfig<ToolVolumesFrom, Boolean> readOnly = new ColumnConfig<>(toolVolumesFromProperties.readOnly(),
+                                                                             appearance.containerVolumesFromReadOnlyWidth(),
+                                                                             appearance.containerVolumesFromReadOnlyLabel());
+
+
+        List<ColumnConfig<ToolVolumesFrom, ?>> columns = new ArrayList<>();
+        columns.add(name);
+        columns.add(namePrefix);
+        columns.add(tag);
+        columns.add(url);
+        columns.add(readOnly);
+        ColumnModel<ToolVolumesFrom> cm = new ColumnModel<>(columns);
+
+        grid = new Grid<>(listStore, cm);
+
+        editing = new GridInlineEditing<>(grid);
```

Comment:
`GridRowEditing` could be more user friendly here. It should probably also be used for the other editable grids in this form.

It may be questionable to use it for the Test Input and Output file grids since they only have 1 column each, but I think the row-editor will capture the `esc` key and simply stop editing the row. With the inline-editor, hitting escape while editing a cell caused the entire form to close and all work in-progress was lost.